docs/overview.rst

```diff
@@ -1,11 +1,410 @@
+
+    * To use only a subset of a dataset's sources, set the ``sources``
+      constructor argument.
+    * You can interact with a dataset via the following methods:
+
+      - :meth:`open`: returns a ``state`` object the dataset will interact with
+        (e.g. a file handle), or ``None`` if it doesn't need to interact with
+        anything.
+      - :meth:`get_data`: given the ``state`` object and an optional ``request``
+        argument, returns data.
+      - :meth:`close`: given the ``state`` object, properly closes it.
+      - :meth:`reset`: given the ``state`` object, properly closes it and
+        returns a fresh one.
+
+    * Use :class:`IterableDataset` to interface with iterable objects in
+      general.
+
+      - :class:`IterableDataset` accepts an ``iterables`` argument, which is
+        expected to be a :class:`dict` mapping from source names to their
+        corresponding iterable objects.
+      - The state :meth:`IterableDataset.open` returns is an iterator object.
+      - Its :meth:`get_data` method doesn't accept requests.
+      - As such, it can only iterate exaplewise and sequentially.
+
+    * Use :class:`IndexableDataset` to interface with indexable objects in
+      general.
+
+      - :class:`IndexableDataset` accepts an ``indexables`` argument, which is
+        expected to be a :class:`dict` mapping from source names to their
+        corresponding indexable objects.
+      - The state :meth:`IndexableDataset.open` returns is ``None``.
+      - Its :meth:`get_data` method expects a list of indices as its
+        ``request`` keyword argument.
+      - As such, it allows random access.
+
+The :class:`Dataset` class is responsible for interfacing with the data and
+handling data access requests. Subclasses of :class:`Dataset` specialize in
+certain types of data.
+
+Datasets contain one or more **sources** of data, such as an array of images,
+a list of labels, a dictionary specifying an ontology, etc. Each source in a
+dataset is identified by a unique name.
+
+IterableDataset
+^^^^^^^^^^^^^^^
+
+The simplest :class:`Dataset` subclass is :class:`IterableDataset`, which
+interfaces with iterable objects.
+
+It is created by passing a :class:`dict` mapping source names to their
+associated data and, optionally, a :class:`dict` mapping source names to tuples
+of axis labels.
+
+>>> from collections import OrderedDict
+>>> from fuel.datasets import IterableDataset
+>>> dataset = IterableDataset(
+...     iterables=OrderedDict([('features', features), ('targets', targets)]),
+...     axis_labels=OrderedDict([('features', ('batch', 'height', 'width')),
+...                              ('targets', ('batch', 'index'))]))
+
+We can ask the dataset what sources of data it provides by accessing its
+``sources`` attribute. We can also know which axes correspond to what by
+accessing its ``axis_labels`` attribute. It also has a ``num_examples`` property
+telling us the number of examples it contains.
```

Comment:
It is my understanding that you can get the `sources`, etc, .. with any :class:`Dataset`, not only with :class:`IterableDataset`. In this case, it would be better to write every common part to the :class:`Dataset` section and keep here only those that are specific to :class:`IterableDataset`, e.g., `iterables`